elastio-nat-provision-lambda/cloudformation-lambda.yaml

```diff
@@ -48,10 +65,17 @@ Resources:
     Properties:
       LogGroupName: /aws/lambda/elastio-nat-gateway-provision
       RetentionInDays: !Ref LambdaLogsRetention
+      Tags:
+        - Key: elastio:resource
+          Value: 'true'
 
   lambdaRole:
     Type: AWS::IAM::Role
     Properties:
+      Tags:
+        - Key: elastio:resource
+          Value: 'true'
```

Comment:
Is it the right call to put our tag on these?  That means `aws-gc` will delete the lambdas.  That seems like it would not be expected behavior...

Reply:
Yeah, it will. We discussed that the Cloudwatch groups should have `elastio:resource` tag https://github.com/elastio/elastio/issues/6694#issuecomment-2261039044, for us to be able to read. I decided to add this tag for all resources anyway. If we do `aws-gc` we'll definitely want to start everything from scratch, likely including the NAT stack as well, because if we need to do `aws-gc` it's probably the case that something in the user's env deletes resources and it already broke both the NAT stack and Connector stack